specutils/manipulation/extract_spectral_region.py

```diff
@@ -75,7 +75,19 @@ def _subregion_to_edge_pixels(subregion, spectrum):
         if not spectral_axis.unit.is_equivalent(u.pix):
             left_index = floor(subregion[0].value)
         else:
-            left_index = int(np.ceil(left_func(subregion).value))
+            if order != "ascending":
+                raise ValueError("u.pix handling should always be ascending")
```

Comment:
```suggestion
                raise ValueError("u.pix spectral axes should always be ascending")
```

This check should probably be in `SpectralAxis.__new__()`, but either way this is clearer.

Reply:
So the order is a local variable calculated in `_subregion_to_edge_pixels` in order to know how to calculate the edge-pixel. Are you suggesting we move the order calculation to `SpectralAxis.__new__` and store that as an attribute that `_subregion_to_edge_pixels` would access?

Reply:
I was thinking keep the ascending/descending check here, but add a similar check in `SpectralAxis` if units are u.pix to raise an error if spectral_axis[-1] < spectral_axis[0] since that is never allowed for u.pix.

Reply:
Then you don't need to check if order is `ascending` in these `u.pix` cases at all, since it's guaranteed.

Reply:
Haha I think I was massively overcomplicating this in my head. Is this what you were imagining? https://github.com/astropy/specutils/pull/1001/commits/efc106b233065996e551235ed201e9db5cd90193